From e7099e3e5a8066b1df057de31d7d3dd4f5725cfe Mon Sep 17 00:00:00 2001
From: ut004146 <zhangqiyi@uniontech.com>
Date: Fri, 27 May 2022 10:36:07 +0800
Subject: [PATCH] fix: fix 'x11perf -shmputxy10' crash in wayland

fix 'x11perf -shmputxy10' crash in wayland

Log: fix 'x11perf -shmputxy10' crash in wayland

Change-Id: I0cec8d70b993e569017f3d9071e79319cabf22e5
Signed-off-by: ut004146 <zhangqiyi@uniontech.com>
---
 hw/xwayland/xwayland-shm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hw/xwayland/xwayland-shm.c b/hw/xwayland/xwayland-shm.c
index ff128316d..a1b085805 100644
--- a/hw/xwayland/xwayland-shm.c
+++ b/hw/xwayland/xwayland-shm.c
@@ -174,7 +174,8 @@ os_create_anonymous_file(off_t size)
     OsBlockSignals();
     do {
         ret = posix_fallocate(fd, 0, size);
-    } while (ret == EINTR);
+    } while (ret == EINTR || ret == EAGAIN);
+    usleep(1);
     OsReleaseSignals();
 
     if (ret != 0) {
-- 
2.20.1

